module-4/backend/week_four.md

```diff
@@ -1,19 +1,41 @@
 ## Week Four - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's your favorite project management tool?
+> From all of the ones that we have tried while at turing I'll say Pivotal tracker is the best one so far.
 2. Why do we create staging environments?
+> It's the closes thing to a production look while still in local, so it's easier to debug.
 3. What are the characteristics of a good README (in your opinion)?
+> When it contains clear and easy to follow instructions on how to install the app,
+> dependencies, migrations, etc.
+> Addiotionally I think it's great when it has what to expect/output from running the app.
+
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+> Accessible through a variety of different technologies, including mobile devices or assistive technologies.
 5. What are some basic security concerns to be aware of when building applications?
+> I will say that as developers we have to keep ourselves updated with OWASP like injection, broken authentication,
+> security misconfiguration, etc.
 6. Why is continuous integration helpful/important?
+>* you can run your tests in the real world
+>* not quite sure, but i think it also increases your code coverage, by telling you what you haven't test.
+>* It deploys to production for you.
 7. What are some continuous integration tools?
+>* Travis CI and recently learn about circle ci.
 
-#### Review  
+#### Review
 
 8. What are some characteristics of "good" git workflow?
+>* commit often
+>* clear and compelling commits.
+>* having good conversations on PRs.
+>* good named feauture branches.
 9. What are the four fundamental concepts of object oriented programming?
+>* Abstraction
+>* Polimorphism
+>* Encapsulation
+>* Inheritance
 10. What's a module in Ruby and what's the difference between a class and a module?
+> A module in Ruby helps to encapsulate code and syntax. It is similar to a class in structure but modules cannot inherit from other modules like a class can inherit from another class.
```

Comment:
what are some other differences between classes and modules? what are some use cases for modules?